hs/CalendarResponse.hs

```diff
@@ -0,0 +1,284 @@
+eventsStr code sect systemTime (startDate, endDate) holidays (start, end)
+    | (startDate == "" || endDate == "") = []
+    | otherwise =
+        ["BEGIN:VEVENT",
+         "DTSTAMP:" ++ systemTime,
+         "DTSTART;TZID=America/Toronto:" ++ startDate ++ start,
+         "DTEND;TZID=America/Toronto:" ++ startDate ++ end,
+         "RRULE:FREQ=WEEKLY;UNTIL=" ++ endDate ++ "000000Z"] ++
+        map (\date -> "EXDATE;TZID=America/Toronto:" ++ date ++ start)
+            holidays ++
+        ["ORGANIZER:University of Toronto",
+         "SUMMARY:" ++ code ++ " " ++ sect,
+         "CATEGORIES:EDUCATION",
+         "END:VEVENT"]
+
+-- ** Ordering data
+
+-- | Orders the time fields for each course; which were obtained from the
+-- database
+orderData :: [Time] -> [[[Double]]]
+orderData courseFields = filter (not . null)
+                                (assignDay $ map timeField courseFields)
+
+-- | Organizes the data by the day of the week, assigning a position based on
+-- the day in the time field.
+assignDay :: [[Double]] -> [[[Double]]]
+assignDay lst = groupBy (\x y -> head x == head y) sortedList
+    where
+        sortedList = sortBy compare lst
+
+-- ** Start time
+
+-- | Obtains the start times for each course by day.
+startTime :: [[[Double]]] -> [[String]]
+startTime dataInOrder =
+    map (\dataByDay -> map getStrTime
+                           (head (sortList dataByDay) :
+                                   getStartConsecutives (sortList dataByDay)))
+        dataInOrder
+    where
+        sortList dataDay = sort $ map (!! 1) dataDay
+
+-- | Gets the start times that are not the very first start time.
+getStartConsecutives :: [Double] -> [Double]
+getStartConsecutives listTimes =
+    filter (/= 30.0) [if listTimes !! i == listTimes !! (i + 1) - 0.5
+                      then 30.0
+                      else listTimes !! (i + 1)| i <- [0 .. lenForComparison]]
+    where
+        lenForComparison = length listTimes - 2
+
+-- ** End time
+
+-- | Obtains the end times for each course by day.
+endTime :: [[[Double]]] -> [[String]]
+endTime dataInOrder =
+    map (\dataByDay -> map (getStrTime . (+ 0.5))
+                           (getEndConsecutives $ sortList dataByDay ++
+                            [last $ sortList dataByDay]))
+        dataInOrder
+    where
+        sortList dataDay = sort $ map (!! 1) dataDay
+
+-- | Gets the end times that are not the very first end time.
+getEndConsecutives :: [Double] -> [Double]
+getEndConsecutives listTimes =
+    filter (/= 30.0) [if listTimes !! i == listTimes !! (i + 1) - 0.5
+                      then 30.0
+                      else listTimes !! i| i <- [0 .. lenForComparison]]
+    where
+        lenForComparison = length listTimes - 2
+
+-- ** Function that works for both start and end times
+
+-- | Creates a string time in the following way HourMinutesSeccondsZ.
+-- For instance, 133500Z corresponds to 1:35 pm.
+getStrTime :: Double -> String
+getStrTime fullTime =
```

Comment:
I have two notes here:
1. Encoding the type of the input/output in a variable/function name is not always good, and generally should be avoided. If the type of the function changes, then so must the name.
2. There is a better way to define a function that gets the string representation of a type in Haskell- can you find which one it is? You will need to change the types to do this.